addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/model/HttpChannelConfig.java

```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    @SuppressWarnings("unused")
+    private @Nullable String stateUrl;
+    @SuppressWarnings("unused")
+    private @Nullable String stateUsername;
+    @SuppressWarnings("unused")
+    private @Nullable String statePassword;
+    private long stateRefreshInterval = DEFAULT_STATE_REFRESH_INTERVAL.toMillis();
+    private long stateConnectTimeout = DEFAULT_CONNECT_TIMEOUT.toMillis();
+    private long stateRequestTimeout = DEFAULT_REQUEST_TIMEOUT.toMillis();
+    @SuppressWarnings("unused")
+    private @Nullable String stateResponseTransform;
+
+    private String commandMethod = DEFAULT_COMMAND_METHOD.name();
+    @SuppressWarnings("unused")
+    private @Nullable String commandUrl;
+    @SuppressWarnings("unused")
+    private @Nullable String commandUsername;
+    @SuppressWarnings("unused")
+    private @Nullable String commandPassword;
+    private long commandConnectTimeout = DEFAULT_CONNECT_TIMEOUT.toMillis();
+    private long commandRequestTimeout = DEFAULT_REQUEST_TIMEOUT.toMillis();
+    private String commandContentType = DEFAULT_CONTENT_TYPE;
+    @SuppressWarnings("unused")
+    private @Nullable String commandRequestTransform;
+    @SuppressWarnings("unused")
+    private @Nullable String commandResponseTransform;
+
+    public Optional<StateRequest> getStateRequest(final BundleContext bundleContext) throws IllegalArgumentException {
+        return Optional.ofNullable(this.stateUrl).map(stateUrl -> {
+            try {
+                final Optional<Transform> stateResponseTransform = Optional.ofNullable(this.stateResponseTransform)
+                        .map(s -> Transform.parse(bundleContext, s));
+                return new StateRequest(
+                        new URL(stateUrl),
+                        Optional.ofNullable(stateUsername),
+                        Optional.ofNullable(statePassword),
+                        Duration.ofMillis(this.stateConnectTimeout),
+                        Duration.ofMillis(this.stateRequestTimeout),
+                        Duration.ofMillis(this.stateRefreshInterval),
+                        stateResponseTransform
+                );
+            } catch (final MalformedURLException e) {
+                throw new IllegalArgumentException("Invalid stateUrl: " + e.getMessage());
+            } catch (final IllegalArgumentException e) {
+                throw new IllegalArgumentException("Invalid stateResponseTransform: " + e.getMessage(), e);
+            }
+        });
+    }
+
+    public Optional<CommandRequest> getCommandRequest(final BundleContext bundleContext) throws IllegalArgumentException {
+        return Optional.ofNullable(this.commandUrl).map(commandUrl -> {
+            final Method commandMethod;
+            try {
+                commandMethod = Method.valueOf(this.commandMethod);
+            } catch (final IllegalArgumentException e) {
+                throw new IllegalArgumentException("Invalid commandMethod", e);
```

Comment:
Why catch and rethrow the same exception, can you create specific ones? Or not catch it?

Reply:
In this case it's to add more information that can be reported to the user about what they need to fix to clear the error.